lib/wal/wal.c

```diff
@@ -651,7 +653,8 @@ wal_open(
     wal->version = WAL_VERSION;
     wal->mp = mp;
     wal->health = health;
-    wal->read_only = rp->read_only;
+    wal->diag_open = (rp->mode == KVDB_MODE_DIAG);
```

Comment:
Can you not just use the same enum here?

Reply:
I will add an interface for this.